### PR TITLE
Updated managed instance swagger and examples

### PR DESCRIPTION
**Pull request by stefanb995**
*Updated ProxyOverride property in swagger file with enum
*Updated examples with ProxyOverride and PublicDataEndpointEnabled

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/examples/ManagedInstanceCreateMax.json [hunk 1]
```diff
@@ -48,7 +50,8 @@
           "subnetId" : "/subscriptions/20D7082A-0FC7-4468-82BD-542694D5042B/resourceGroups/testrg/providers/Microsoft.Network/virtualNetworks/vnet1/subnets/subnet1",
           "state": "Ready",
           "collation": "SQL_Latin1_General_CP1_CI_AS",
-          "dnsZone": "1b4e2caff2530"
+          "dnsZone": "1b4e2caff2530",
```
Comment (jaredmoo) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/examples/ManagedInstanceCreateMax.json]: Is publicDataEndpointEnabled intentionally not there in the response?

File: specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/examples/ManagedInstanceListByResourceGroup.json [hunk 2]
```diff
@@ -47,7 +49,8 @@
                         "storageSizeInGB": 1024,
                         "licenseType": "Full",
                         "collation":  "SQL_Latin1_General_CP1_CI_AS",
-                        "dnsZone": "1b4e2caff2530"
+                        "dnsZone": "1b4e2caff2530",
+                        "publicDataEndpointEnabled" :  "false"
```
Comment (jaredmoo) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/examples/ManagedInstanceListByResourceGroup.json]: Is proxyOverride intentionally missing?

File: specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/managedInstances.json [hunk 3]
```diff
@@ -371,6 +371,19 @@
...
+          "description": "Whether or not the public data endpoint is enabled.",
+          "type": "boolean"
+        },
+        "proxyOverride": {
+          "description": "Connection type used for connecting to the instance.",
+          "enum": ["Proxy", "Redirect", "Default"],
+          "x-ms-enum":{
+            "name": "ManagedInstanceProxyOverride",
+            "modelAsString": true
+		  },
```
Comment (jaredmoo) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/managedInstances.json]: Nit: please fix indentation

**Comment by veronicagg** [comment]
@anuchandy giving you this PR as I took your other one (just a swap) thanks!

File: specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/examples/ManagedInstanceCreateMax.json (hunk 1 above)
Comment (stefanb995) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/examples/ManagedInstanceCreateMax.json]: Yes, it's by design.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/examples/ManagedInstanceListByResourceGroup.json (hunk 2 above)
Comment (stefanb995) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/examples/ManagedInstanceListByResourceGroup.json]: This is the case when the value for that column is null (instance created before introducing new property).

File: specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/managedInstances.json (hunk 3 above)
Comment (stefanb995) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/managedInstances.json]: Fixed.

**Comment by anuchandy** [comment]
@stefanb995 please fix conflicts in managedInstances.json.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/examples/ManagedInstanceCreateMax.json [hunk 4]
```diff
@@ -23,7 +23,9 @@
         "storageSizeInGB": 1024,
         "licenseType": "Full",
         "collation": "SQL_Latin1_General_CP1_CI_AS",
-        "dnsZonePartner": "/subscriptions/20D7082A-0FC7-4468-82BD-542694D5042B/resourceGroups/testrg/providers/Microsoft.Sql/managedInstances/testinstance"
+        "dnsZonePartner": "/subscriptions/20D7082A-0FC7-4468-82BD-542694D5042B/resourceGroups/testrg/providers/Microsoft.Sql/managedInstances/testinstance",
+        "publicDataEndpointEnabled" : "false",
```
Comment (nschonni) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/examples/ManagedInstanceCreateMax.json]: Don't think the boolean value should be quoted

Comment (stefanb995) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/examples/ManagedInstanceCreateMax.json]: Thanks. I fixed it.

**Review by jaredmoo — APPROVED** [review]

**Review by anuchandy — CHANGES_REQUESTED** [review]
@stefanb995 could you fix the conflict in managedInstances.json

**Review by jaredmoo — APPROVED** [review]

**Comment by stefanb995** [comment]
Can you please approve this. I need it merged to regenerate SDK from master branch.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/examples/ManagedInstanceCreateMax.json (hunk 1 above)
Comment (stefanb995) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/examples/ManagedInstanceCreateMax.json]: I talked with Mladen who was implementing this feature. He actually forgot to include it here. It should be in response. I'll update it and he'll make sure to update his part.

**Comment by anuchandy** [comment]
@stefanb995 it seems there are still conflicts in managedInstances.json?

**Comment by stefanb995** [comment]
My bad. It should be fixed now.

**Review by anuchandy — APPROVED** [review]

**Pull request merged**